Tests/TuistKitTests/Generator/ProjectGeneratorTests.swift

```diff
@@ -33,4 +33,66 @@ final class ProjectGeneratorTests: XCTestCase {
                                    system: system,
                                    resourceLocator: resourceLocator)
     }
+
+    func test_generate_defaultGroupsOrder() throws {
+        // Given
+        let project = Project.test()
+
+        // When
+        let path = try subject.generate(project: project,
+                                        options: GenerationOptions(),
+                                        graph: Graph.test(),
+                                        sourceRootPath: fileHandler.currentPath)
+
+        // Then
+        let xcodeproj = try XcodeProj(pathString: path.path.asString)
+        let mainGroup = try xcodeproj.pbxproj.rootGroup()
+        XCTAssertEqual(mainGroup?.children.map { $0.nameOrPath }, [
+            "Project",
+            "Manifest",
+            "Frameworks",
+            "Products",
+        ])
+    }
+
+    func test_generate_customGroupsOrder() throws {
```

Comment:
So this feels a bit repetitive test wise, as this logic is mostly tested in the other tests - however the bit that isn't covered is an integration style test that ensures the generated project does indeed have the groups sorted.